### PR TITLE
[ALLUXIO-1907] fix and unignore free tests

### PR DESCRIPTION
**Pull request by siyangy**
The BlockRemover in BlockWorker is executed asynchronously, so in test we need to wait for all of the involved blocks being removed.

### CODE REVIEW

**Comment by siyangy** [comment]
jenkins, test this please

**Comment by yupeng9** [comment]
jenkins, test this please

**Comment by siyangy** [comment]
jenkins, test this please

**Comment by siyangy** [comment]
@yupeng9, this is probably ready for review. I tried rerunning build a few times after the fix and the only failure was lineageRecoveryTest, which might be a different flaky test.

File: tests/src/test/java/alluxio/client/FreeAndDeleteIntegrationTest.java [hunk 1]
```diff
@@ -128,9 +121,14 @@ public void freeAndDeleteIntegrationTest() throws Exception {
...
 
     mFileSystem.delete(filePath);
-    // File is immediately gone after delete.
-    mThrown.expect(FileDoesNotExistException.class);
-    mFileSystem.getStatus(filePath);
+
+    try {
+      // File is immediately gone after delete.
+      mFileSystem.getStatus(filePath);
+      Assert.fail();
```
Comment (yupeng9) [inline tests/src/test/java/alluxio/client/FreeAndDeleteIntegrationTest.java]: Can you add a failure message?

**Comment by yupeng9** [comment]
Thanks for the fix! Could you briefly describe what caused the flakiness?

**Comment by siyangy** [comment]
Updated the flakiness reason in PR description.

File: tests/src/test/java/alluxio/client/FreeAndDeleteIntegrationTest.java (hunk 1 above)
Comment (siyangy) [inline tests/src/test/java/alluxio/client/FreeAndDeleteIntegrationTest.java]: done

File: tests/src/test/java/alluxio/shell/command/FreeCommandTest.java [hunk 2]
```diff
@@ -33,58 +32,64 @@
...
     try {
       // Schedule 1st heartbeat from worker.
       Assert.assertTrue(HeartbeatScheduler.await(HeartbeatContext.WORKER_BLOCK_SYNC, 5,
           TimeUnit.SECONDS));
       HeartbeatScheduler.schedule(HeartbeatContext.WORKER_BLOCK_SYNC);
 
       // Waiting for the removal of blockMeta from worker.
-      while (mLocalAlluxioCluster.getWorker().getBlockWorker().hasBlockMeta(blockId)) {
-        CommonUtils.sleepMs(50);
+      for (long blockId : blockIds) {
```
Comment (yupeng9) [inline tests/src/test/java/alluxio/shell/command/FreeCommandTest.java]: Can you use `CommonTestUtils.waitFor` instead?

**Comment by yupeng9** [comment]
Thanks. I left one comment.

File: tests/src/test/java/alluxio/shell/command/FreeCommandTest.java (hunk 2 above)
Comment (siyangy) [inline tests/src/test/java/alluxio/shell/command/FreeCommandTest.java]: Done. It's not very clear to me why CommonTestUtils.waitFor is more in favor though, can you talk about that a bit more?

**Comment by yupeng9** [comment]
LGTM. It's more for consistency of the utility function use.

**Pull request merged**